web/handlers/admin/settings.go

```diff
@@ -78,6 +90,44 @@ func (h settingsHandler) setPrivacy(w http.ResponseWriter, req *http.Request) {
 	}
 
 	// we successfully set the privacy mode! time to redirect to the updated settings overview
-	overview := h.urlTo(router.AdminSettings).String()
-	http.Redirect(w, req, overview, http.StatusFound)
+	h.redirect(router.AdminSettings, w, req)
+}
+
+/* common-use functions */
+
+func (h settingsHandler) getMember(w http.ResponseWriter, req *http.Request) *roomdb.Member {
+	// get the member behind the POST
+	currentMember := members.FromContext(req.Context())
+	if currentMember == nil {
+		err := weberrors.ErrForbidden{Details: fmt.Errorf("not a registered member")}
+		h.r.Error(w, req, http.StatusInternalServerError, err)
+		return nil
+	}
+
+	// make sure the member is an admin
+	if currentMember.Role != roomdb.RoleAdmin {
+		err := weberrors.ErrForbidden{Details: fmt.Errorf("yr not an admin! naughty naughty")}
+		h.r.Error(w, req, http.StatusInternalServerError, err)
+		return nil
+	}
+	return currentMember
+}
+
+func (h settingsHandler) verifyPostRequirements(w http.ResponseWriter, req *http.Request) bool {
+	if req.Method != "POST" {
+		// TODO: proper error type
+		h.r.Error(w, req, http.StatusBadRequest, fmt.Errorf("bad request"))
```

Comment:
```suggestion
		err := weberrors.ErrBadRequest{Where: "HTTP Method", Details: fmt.Errorf("expected POST not %s", req.Method)}
		h.r.Error(w, req, http.StatusBadRequest, err)
```

these landed in #115. :)

ps: i like this helper, could possible detatch it from `settingsHelper` and use it in the other handlers, too but let's do that in a later refactor.